File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/determine-whether-matrix-can-be-obtained-by-rotation.java
class Solution {
    public boolean findRotation(int[][] mat, int[][] target) {
        int n = mat.length;
        if(Arrays.deepEquals(mat,target)) return true;

        int res1[][] = new int[n][n];

        //first 90

        for(int i =0;i<n;i++){
            for(int j=0;j<n;j++){
                res1[i][j]=mat[n-j-1][i];
            }
        }
        if(Arrays.deepEquals(res1,target)) return true;

        int res2[][] = new int[n][n];
        for(int i =0;i<n;i++){
            for(int j = 0;j<n;j++){
                res2[i][j] = res1[n-j-1][i];
            }
        }
        if(Arrays.deepEquals(res2,target)) return true;
        int res3[][] = new int[n][n];
        for(int i =0;i<n;i++){
            for(int j = 0;j<n;j++){
                res3[i][j] = res2[n-j-1][i];
            }
        }
        if(Arrays.deepEquals(res3,target)) return true;





        return false;



        
    }
}

File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/determine-whether-matrix-can-be-obtained-by-rotation.ts
function findRotation(mat: number[][], target: number[][]): boolean {
    function rotate(matrix: number[][]): number[][] {
        const n = matrix.length;
        const rotated = Array.from({ length: n }, () => Array(n).fill(0));

        for (let i = 0; i < n; i++) {
            for (let j = 0; j < n; j++) {
                rotated[j][n - 1 - i] = matrix[i][j];
            }
        }

        return rotated;
    }

    let current = mat;

    // Check original orientation
    if (JSON.stringify(current) === JSON.stringify(target)) return true;

    // Check 3 more rotations
    for (let i = 1; i < 4; i++) {
        current = rotate(current);
        if (JSON.stringify(current) === JSON.stringify(target)) return true;
    }

    return false;
}